test/binaryen.js/expressionrunner.js

```diff
@@ -0,0 +1,156 @@
+var Flags = binaryen.ExpressionRunner.Flags;
+console.log("// ExpressionRunner.Flags.Default = " + Flags.Default);
+console.log("// ExpressionRunner.Flags.PreserveSideeffects = " + Flags.PreserveSideeffects);
+console.log("// ExpressionRunner.Flags.TraverseCalls = " + Flags.TraverseCalls);
+
+binaryen.setAPITracing(true);
+
+var module = new binaryen.Module();
+module.addGlobal("aGlobal", binaryen.i32, true, module.i32.const(0));
+
+// Should evaluate down to a constant
+var runner = new binaryen.ExpressionRunner(module);
+var expr = runner.runAndDispose(
+  module.i32.add(
+    module.i32.const(1),
+    module.i32.const(2)
+  )
+);
+assert(JSON.stringify(binaryen.getExpressionInfo(expr)) === '{"id":14,"type":2,"value":3}');
+
+// Should traverse control structures
+runner = new binaryen.ExpressionRunner(module);
+expr = runner.runAndDispose(
+  module.i32.add(
+    module.i32.const(1),
+    module.if(
+      module.i32.const(0),
+      module.i32.const(0),
+      module.i32.const(3)
+    )
+  ),
+);
+assert(JSON.stringify(binaryen.getExpressionInfo(expr)) === '{"id":14,"type":2,"value":4}');
+
+// Should be unable to evaluate a local if not explicitly specified
+runner = new binaryen.ExpressionRunner(module);
+expr = runner.runAndDispose(
+  module.i32.add(
+    module.local.get(0, binaryen.i32),
+    module.i32.const(1)
+  )
+);
+assert(expr === 0);
```

Comment:
Would it be more idiomatic for the JS API to turn this into `null` or something like that?

Reply:
Figured that one would typically test this as `!expr` anyway, just doing an overly precise check here for testing purposes. Would imagine that not mixing `0` and `null` has benefits for the JIT.

Reply:
An alternative here is to return the unmodified original expression. Would that be an improvement?